src/settings.test.js

```diff
@@ -0,0 +1,136 @@
+import { expect } from 'chai';
+import _ from 'lodash';
+import settings from './settings';
+import utils from './utils';
+
+
+describe('/settings', () => {
+  describe('.set()', () => {
+    describe('path is a string', () => {
+      it('sets chosen parameters with current values', () => {
+        const testSettings = _.cloneDeep(settings);
+        testSettings.set('modes.VD.kSigma', 3);
+        expect(testSettings.now.modes.VD.kSigma).to.be.equal(3);
+      });
+    });
+
+    describe('path is not a string (is an object)', () => {
+      it('makes diff with settings.now and sets diff', () => {
+        const testSettings = _.cloneDeep(settings);
+
+        const testDiffs = _.cloneDeep(settings);
+        testDiffs.set('modes.VD.kSigma', 3);
+        testSettings.set(testDiffs.now);
+        expect(testSettings.now.modes.VD.kSigma).to.be.equal(3);
+      });
+    });
+  });
+
+  describe('.get()', () => {
+    it('gets current parameter', () => {
+      const testSettings = _.cloneDeep(settings);
+      testSettings.now.modes.VD.kSigma = 1;
+      expect(settings.get('modes.VD.kSigma')).to.be.equal(1);
+    });
+  });
+
+  describe('.reset()', () => {
+    it('resets all parameters in default', () => {
+      const testSettings = _.cloneDeep(settings);
+
+      testSettings.set('modes.VD.kSigma', 3);
+
+      testSettings.reset();
+      expect(testSettings).to.be.eql(settings);
+    });
+  });
+
+  describe('.checkpoint()', () => {
+    it('saved "now" parameters in "old" field', () => {
+      const testSettings = _.cloneDeep(settings);
+      testSettings.now.modes.VD.kSigma = 3;
+      testSettings.checkpoint();
+      expect(testSettings.now.modes.VD.kSigma).to.be.equal(testSettings.old.modes.VD.kSigma);
```

Comment:
Again, do not write to `now` directly **anywhere**, use `set()` method. Also:
- maybe we should compare the entire objects (as stated in the test title)?
- maybe we should check for the exact value (to distinguish from `reset()`)?

Reply:
i've checked the entire objects as checkpoint() preserve the whole structure